wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -153,7 +153,7 @@ public void refreshToken(@Parameter(description = "Factory url") @QueryParam("ur
               singletonMap(URL_PARAMETER_NAME, url));
       if (!authorisationRequestManager.isStored(factoryParametersResolver.getProviderName())) {
         personalAccessTokenManager.getAndStore(
-            factoryParametersResolver.parseFactoryUrl(url).getHostName());
+            factoryParametersResolver.parseFactoryUrl(url).getProviderUrl());
```

Comment:
@vinokurig this looks like the main change of the PR, right? My understanding tha the rest is improved test coverage / refactoring. Maybe we need some comment / javadoc here?

Reply:
> this looks like the main change of the PR, right?

Not really, the other fix is to throw an unauthorised exception with an authorisation url instead of `ScmCommunicationException` if an oauth token is not valid in all the scm implementations.

Reply:
> Maybe we need some comment / javadoc here?

done